benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -0,0 +1,228 @@
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    prelude::{Added, Changed},
+    world::World,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::{prelude::SliceRandom, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+criterion_group!(
+    benches,
+    all_added_detection,
+    all_changed_detection,
+    few_changed_detection,
+    none_changed_detection,
+);
+criterion_main!(benches);
+
+#[derive(Component, Default)]
+#[component(storage = "Table")]
+struct Table;
+#[derive(Component, Default)]
+#[component(storage = "SparseSet")]
+struct Sparse;
+
+const RANGE_ENTITIES_TO_BENCH_COUNT: std::ops::Range<u32> = 5..7;
+
+type BenchGroup<'a> = criterion::BenchmarkGroup<'a, criterion::measurement::WallTime>;
+
+fn deterministic_rand() -> ChaCha8Rng {
+    ChaCha8Rng::seed_from_u64(42)
+}
+
+fn setup<T: Component + Default>(entity_count: u32) -> World {
+    let mut world = World::default();
+    world.spawn_batch((0..entity_count).map(|_| (T::default(),)));
+    black_box(world)
+}
+
+fn generic_bench<P: Copy>(
+    bench_group: &mut BenchGroup,
+    mut benches: Vec<Box<dyn FnMut(&mut BenchGroup, P)>>,
+    bench_parameters: P,
+) {
+    for b in &mut benches {
+        b(bench_group, bench_parameters);
+    }
+}
```

Comment:
generic helper to avoid duplication, see https://github.com/bevyengine/bevy/issues/5161.